keras_dqn/agent.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+from keras.utils.np_utils import to_categorical
+
+
+class Agent(object):
+    def __init__(self, model, target_model=None, target_swap=None):
+        """
+        Expected values in kwargs:
+
+        "update_target_every" : number of frames skipped before syncing the target with model
+        "update_model_every" : number of frames skipped before gradient step
+        """
+        self.model = model
+        if target_model is None:
+            self.target_model = model
+        else:
+            self.target_model = target_model
+            self.target_swap = True
+            print("Using target model update.")
+
+    @property
+    def num_actions(self):
+        return self.model.output_shape[-1]
+
+    def train_on_batch(self, state, action, reward, next_state, terminal, gamma=.99, accum_mode="mean"):
```

Comment:
can we include the shape you are expecting here? I can make the Memory module obey what we decide on here